### PR TITLE
Feature/12102 implement logic for g status change

### PR DESCRIPTION
**Pull request by 30mar**
## Description
add logic for the new G status update

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-12102

## Screenshots
https://user-images.githubusercontent.com/15270737/156960733-62d38f07-5dbc-4b3e-83cf-12474be34fe1.MP4

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/HealthCertificatesTabCoordinator.swift [hunk 1]
```diff
@@ -281,6 +281,9 @@ final class HealthCertificatesTabCoordinator {
 			presentCovPassInfoScreen: { [weak self] viewController in
 				self?.presentCovPassInfoScreen(rootViewController: viewController)
 			},
+			onDismiss: { [weak self] in
+				self?.overviewScreen.tableView.reloadData()
+			},
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/HealthCertificatesTabCoordinator.swift:286]: Not needed if you use the same `objectDidChange` mechanism that we use for `isNewBoosterRule` and `isNewCertificateReissuance`

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService.swift [hunk 2]
```diff
@@ -603,11 +603,14 @@ class HealthCertificateService: HealthCertificateServiceServable {
...
 			case .success(let dccWalletInfo):
+				let isAdmissionStateChanged = dccWalletInfo.admissionState != person.dccWalletInfo?.admissionState
 				let previousBoosterNotificationIdentifier = person.boosterRule?.identifier ?? person.dccWalletInfo?.boosterNotification.identifier
 				let previousCertificateReissuance = person.dccWalletInfo?.certificateReissuance
 				person.dccWalletInfo = dccWalletInfo
 				person.mostRecentWalletInfoUpdateFailed = false
-				
+				if isAdmissionStateChanged {
+					person.isAdmissionStateChanged = isAdmissionStateChanged
+				}
```
[the remark this answers is not in the document]
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService.swift:613]: And a question: is the notification that would be triggered here handled in a separate subtask?

[the remark this answers is not in the document]
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService.swift:613]: Second question: is showing the new `stateChangeNotificationText` property text on the DCCWalletInfo part of a different subtask as well? I only see another task for unit and ui tests, that's why I'm asking.

**Review by nickguendling — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/HealthCertifiedPerson/Cells/AdmissionState/AdmissionStateCellModel.swift [hunk 3]
```diff
@@ -29,7 +29,12 @@ final class AdmissionStateCellModel {
 	}
 
 	var description: String? {
-		healthCertifiedPerson.dccWalletInfo?.admissionState.longText?.localized(cclService: cclService)
+		if healthCertifiedPerson.isAdmissionStateChanged {
+			return "Ihr Status hat sich geändert. Ihre Zertifikate erfüllen jetzt die 2G-Regel. Wenn Sie Ihren aktuellen Status vorweisen müssen, schließen Sie diese Ansicht und zeigen Sie den QR-Code auf der Zertifikatsübersicht.\nMehr Informationen zu Ihrem Status im FAQ."
+			// return healthCertifiedPerson.dccWalletInfo?.admissionState.stateChangeNotificationText?.localized(cclService: cclService)
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/HealthCertifiedPerson/Cells/AdmissionState/AdmissionStateCellModel.swift:34]: I think this was just meant for testing purposes :D

**Review by nickguendling — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/HealthCertifiedPerson/Cells/AdmissionState/AdmissionStateCellModel.swift (hunk 3 above)
Comment (30mar) [reply src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/HealthCertifiedPerson/Cells/AdmissionState/AdmissionStateCellModel.swift]: yes, thanks for noticing, i pushed again

**Review by pascal-brause — APPROVED** [review]
If you fixed Nick's changes, it is fine for me

**Review by nickguendling — APPROVED** [review]

**Review by naveeddotio — APPROVED** [review]

**Pull request merged**